Packs/CiscoFirepower/Integrations/CiscoFirepower/CiscoFirepower.py

```diff
@@ -1469,120 +3606,845 @@ def get_task_status_command(client: Client, args: Dict) -> Tuple[str, Dict, Dict
+    """
+    Creates or overrides the Snort3 Intrusion rule group with the specified parameters.
+
+    Args:
+        client (Client): Session to Cisco Firepower Management Center to run desired requests.
+        args (Dict[str, Any]): Arguments passed down by the CLI to provide in the HTTP request.
+
+    Returns:
+        CommandResults: Information about the created Intrusion Rule.
+    """
+    rule_data = args['rule_data']
+    rule_group_ids = argToList(args['rule_group_ids'])
+
+    raw_response = client.create_intrusion_rule(
+        rule_data=rule_data,
+        rule_group_ids=rule_group_ids,
+    )
+
+    return parse_results(
+        raw_response=raw_response,
+        command_headers_by_keys=INTRUSION_RULE_HEADERS_BY_KEYS,
+        command_title=f'Created {INTRUSION_RULE_TITLE}',
+        command_context=INTRUSION_RULE_CONTEXT,
+    )
+
+
+def list_intrusion_rule_command(client: Client, args: Dict[str, Any]) -> CommandResults:
+    """
+    Retrieves the Snort3 Intrusion rule group.
+    If no ID is specified, retrieves a list of all Snort3 Intrusion rule groups.
+    - GET argument: intrusion_rule_id.
+    - LIST arguments: sort, filter, expanded_response, limit, page, page_size.
+
+    Args:
+        client (Client): Session to Cisco Firepower Management Center to run desired requests.
+        args (Dict[str, Any]): Arguments passed down by the CLI to provide in the HTTP request.
+
+    Raises:
+        ValueError: In case the user has entered both GET and LIST arguments, raise an error.
+
+    Returns:
+        CommandResults: Information about Intrusion Rules.
+    """
+    # GET arguments
+    intrusion_rule_id = args.get('intrusion_rule_id', '')
+    # LIST arguments
+    limit = arg_to_number(args.get('limit', 0))
+    page = arg_to_number(args.get('page', 0))
+    page_size = arg_to_number(args.get('page_size', 0))
+    sort = argToList(args.get('sort'))
+    filter_string = args.get('filter')
+    expanded_response = argToBoolean(args.get('expanded_response', 'False'))
+
+    is_get_request = check_is_get_request(
+        get_args=[intrusion_rule_id],
+        list_args=[sort, filter_string, expanded_response, limit, page, page_size]
+    )
+
+    if is_get_request:
+        raw_response = client.get_intrusion_rule(
+            intrusion_rule_id=intrusion_rule_id,
+        )
+
+    else:  # is_list_request
+        raw_response, raw_responses = client.list_intrusion_rule(
+            page=page,
+            page_size=page_size,
+            limit=limit,
+            sort=sort,
+            filter_string=filter_string,
+            expanded_response=expanded_response,
+        )
+
+    command_results = parse_results(
+        raw_response=raw_response,
+        command_headers_by_keys=INTRUSION_RULE_HEADERS_BY_KEYS,
+        command_title=f'Fetched {INTRUSION_RULE_TITLE}',
+        command_context=INTRUSION_RULE_CONTEXT,
+    )
+
+    if not is_get_request:
+        command_results.raw_response = raw_responses
```

Comment:
why not do it for get requests?

Reply:
In the commands there are two function calls, the functions handle `GET` and `LIST` request according to the user's input.

In all of the list requests I return a `Tuple` that holds two arguments, the first is the appended `items` within all the responses and the second is the the actual `raw response`
All the list requests are handled in a regular manner (single request), therefore only a `raw_response` is returned.

Reply:
Let's move this part to row 3857 (as part of the `if` block), to keep the flow clear.
By the way, we can remove the `is_get_request` variable at all - in line 3854 just do `if check_is_get_requect(...):`

Reply:
please apply it in all functions, I see it's a recurring thing. The flow is a bit confusing this way

Reply:
The thing is `parse_results` has to be called after the `if` and `else` statements.
I can implement this suggestion If I hold an empty variable `raw_responses = None` and also add this argument to the `parse_results` method which will set it if it exists.
What do you think?

You may see the new implementation below, also the `command_results.raw_response` has passed to the method.
```
raw_responses = None

    if check_is_get_request(
        get_args=[intrusion_rule_id],
        list_args=[sort, filter_string, expanded_response, limit, page, page_size]
    ):
        raw_response = client.get_intrusion_rule(
            intrusion_rule_id=intrusion_rule_id,
        )

    else:  # is_list_request
        raw_response, raw_responses = client.list_intrusion_rule(
            page=page,
            page_size=page_size,
            limit=limit,
            sort=sort,
            filter_string=filter_string,
            expanded_response=expanded_response,
        )

    command_results = parse_results(
        raw_response=raw_response,
        command_headers_by_keys=INTRUSION_RULE_HEADERS_BY_KEYS,
        command_title=f'Fetched {INTRUSION_RULE_TITLE}',
        command_context=INTRUSION_RULE_CONTEXT,
        raw_responses=raw_responses,
    )
```